opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/WebOpacAt.java

```diff
@@ -0,0 +1,290 @@
+ * is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+package de.geeksfactory.opacclient.apis;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import de.geeksfactory.opacclient.objects.Detail;
+import de.geeksfactory.opacclient.objects.DetailledItem;
+import de.geeksfactory.opacclient.objects.Filter;
+import de.geeksfactory.opacclient.objects.Library;
+import de.geeksfactory.opacclient.objects.SearchRequestResult;
+import de.geeksfactory.opacclient.objects.SearchResult;
+import de.geeksfactory.opacclient.searchfields.SearchField;
+import de.geeksfactory.opacclient.searchfields.SearchQuery;
+import de.geeksfactory.opacclient.searchfields.TextSearchField;
+
+/**
+ * An implementation of {@link *.web-opac.at} sites operated by the Austrian company
+ * <a href="https://littera.eu/">Littera</a>.
+ */
+public class WebOpacAt extends ReadOnlyApi {
+    protected static final Map<String, String> LANGUAGE_CODES = new HashMap<String, String>() {{
+        put("en", "eng");
+        put("de", "deu");
+        //put("tr", "tur");
+    }};
+    protected static final Map<String, SearchResult.MediaType> MEDIA_TYPES =
+            new HashMap<String, SearchResult.MediaType>() {{
+        // de
+        put("Book", SearchResult.MediaType.BOOK);
+        put("Zeitschrift", SearchResult.MediaType.MAGAZINE);
+        put("CD ROM", SearchResult.MediaType.CD_SOFTWARE);
+        put("DVD", SearchResult.MediaType.DVD);
+        put("Hörbuch", SearchResult.MediaType.AUDIOBOOK);
+        put("eMedium", SearchResult.MediaType.EBOOK);
+        // en
+        put("Book", SearchResult.MediaType.BOOK);
+        put("Periodical", SearchResult.MediaType.MAGAZINE);
+        put("CD ROM", SearchResult.MediaType.CD_SOFTWARE);
+        put("DVD", SearchResult.MediaType.DVD);
+        put("Audiobook", SearchResult.MediaType.AUDIOBOOK);
+        put("eMedium", SearchResult.MediaType.EBOOK);
+    }};
```

Comment:
Is there any way of detecting the media types independent from the language? In most other APIs we use the file names of media type icons that are shown in each search result or CSS classes.
If that's not possible, @raphaelm, what would you think about an API independent thing for detecting media types from their names, similar to the `MeaningDetector` for search fields? This would also be helpful for the `Open` API because some libraries do not have media type icons.